src/dbnode/client/circuitbreaker/middleware/enable_provider.go

```diff
@@ -0,0 +1,109 @@
+package middleware
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/m3db/m3/src/cluster/kv"
+	"go.uber.org/zap"
+)
+
+// EtcdConfig represents the configuration stored in etcd.
+type EtcdConfig struct {
```

Comment:
nits:
- we don't call these types `Etcd...`
- use the proto generation for any types needing the proto tags - they're put here in this repo https://github.com/m3db/m3/tree/master/src/dbnode/generated/proto